core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
```

Comment:
I can't see an increase in RAM: For me it is 1.08 GB -> 1.05 GB Wow, that's pretty much! We definitely need to cut that down somehow.